detectable/src/main/java/com/synopsys/integration/detectable/detectables/yarn/parse/entry/YarnLockEntryParser.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * detectable
+ *
+ * Copyright (c) 2021 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.detectable.detectables.yarn.parse.entry;
+
+import java.util.List;
+import java.util.Optional;
+
+import com.synopsys.integration.detectable.detectables.yarn.parse.YarnLockLineAnalyzer;
+import com.synopsys.integration.detectable.detectables.yarn.parse.entry.element.YarnLockElementParser;
+
+public class YarnLockEntryParser {
+    private final YarnLockLineAnalyzer yarnLockLineAnalyzer;
+    private final YarnLockElementParser yarnLockEntryElementParser;
+
+    public YarnLockEntryParser(YarnLockLineAnalyzer yarnLockLineAnalyzer, YarnLockElementParser yarnLockEntryElementParser) {
+        this.yarnLockLineAnalyzer = yarnLockLineAnalyzer;
+        this.yarnLockEntryElementParser = yarnLockEntryElementParser;
+    }
+
+    public YarnLockEntryParseResult parseEntry(List<String> yarnLockFileLines, int entryStartIndex) {
```

Comment:
I think if these were named more like 'parseNextEntry' or 'parseNextLine' it would help clarify what is happening.